File: ui/src/packages/flow-typed/checkers/model/checkers.js
// @flow
import type { CheckersBoard } from './type/checkersBoard';
import type { PlayerNb } from './type/player';
import type { Id } from './type/id';
import type { Token } from './type/token';

export type Checkers = $ReadOnly<{|
  id: Id,
  name: string,
  board: CheckersBoard,
  player: {
    id: Id,
    name: string,
    playerNb: PlayerNb
  },
  turnPlayer: PlayerNb,
  winner: ?PlayerNb,
  token: Token,
|}>;

File: api/checkers/checkers-domain/src/main/java/io/github/anthogdn/iataaa/checkersDomain/model/Checkers.java
package io.github.anthogdn.iataaa.checkersDomain.model;

public class Checkers {
    private CheckersBoard checkersBoard;
    private PlayerNb turnPlayer;
    private PlayerNb winner;

    public Checkers(CheckersBoard checkersBoard, PlayerNb turnPlayer, PlayerNb winner) {
        this.checkersBoard = checkersBoard;
        this.turnPlayer = turnPlayer;
        this.winner = winner;
    }

    public CheckersBoard getCheckersBoard() {
        return checkersBoard;
    }
    public void setCheckersBoard(CheckersBoard checkersBoard) {
        this.checkersBoard = checkersBoard;
    }

    public PlayerNb getTurnPlayer() {
        return turnPlayer;
    }
    public void setTurnPlayer(PlayerNb turnPlayer) {
        this.turnPlayer = turnPlayer;
    }

    public PlayerNb getWinner() {
        return winner;
    }
    public void setWinner(PlayerNb winner) {
        this.winner = winner;
    }
}
